### DISCUSSION TITLE
Have actions *not* run a job if a label is set

### DISCUSSION THREAD

**Discussion by mathomp4** [community/community#26712] [Actions]
<p>All,</p>
<p>I think I’m an idiot as I can’t seem to figure this out. Namely, a repo I work with can have both trivial and non-trivial PRs. We have our developers label the trivial ones with a label “0 diff trivial”. What I’d like is to have those PRs <em>not</em> run a rather expensive CI step (takes 30 minutes to build) as changing a README file shouldn’t need that.</p>
<p>So following some other threads here I have something a la:</p>
<pre><code class="lang-yaml">name: Build Tests

on:
  pull_request:
    types: [opened, synchronize, reopened, labeled, unlabeled]

jobs:
  build_cheap:
    name: Build Cheap
    runs-on: ubuntu-latest
    container: foo
    steps:
      - name: Cancel Previous Runs
        uses: styfle/cancel-workflow-action@0.4.0
        with:
          access_token: ${{ github.token }}
      - name: Checkout
        uses: actions/checkout@v2
...
  build_expensive:
    name: Build expensive
    if: "!contains(github.event.pull_request.labels.*.name, 'trivial')"
    runs-on: ubuntu-latest
    container: foo
    steps:
      - name: Cancel Previous Runs
        uses: styfle/cancel-workflow-action@0.4.0
        with:
          access_token: ${{ github.token }}
      - name: Checkout GCM
        uses: actions/checkout@v2
</code></pre>
<p>My guess is my problem is with not knowing GitHub Actions if syntax. I thought:</p>
<pre><code class="lang-yaml">    if: "!contains(github.event.pull_request.labels.*.name, 'trivial')"
</code></pre>
<p>was the right way to say “Run this job step only if the labels do not have trivial”, but tests seem to show this causes an expensive test. :confused:</p>
<p>(Note: I might have other issues as well. Since I care about labels, I added to the types of the <code>on: pull_request</code>. I also tried adding the “Cancel Previous Runs” action since I figured we sometimes have to push to our PRs. But I look and it’s not cancelling so…)</p>

**Comment by mathomp4** [comment, marked as answer]
<p>Sigh. I might have figured this out. I actually decided to read the documentation carefully and:</p>
<blockquote>
<p><code>contains( search, item )</code></p>
<p>Returns  <code>true</code>  if  <code>search</code>  contains  <code>item</code> . If  <code>search</code>  is an array, this function returns  <code>true</code>  if the  <code>item</code>  is an element in the array. If  <code>search</code>  is a string, this function returns  <code>true</code>  if the  <code>item</code>  is a substring of  <code>search</code> . This function is not case sensitive. Casts values to a string.</p>
</blockquote>
<p>After staring at the GitHub API docs, <code>labels</code> is an array, which means contains seems to be looking for a “full string” (as it were) not just something <em>in</em> the string. Thus:</p>
<pre><code class="lang-yaml">    if: "!contains(github.event.pull_request.labels.*.name, '0 diff trivial')"
</code></pre>
<p>seems more correct.</p>

**Comment by craigfay** [comment]
<p>Try wrapping your condition with the double <code>${{ }}</code> <a href="https://docs.github.com/en/actions/learn-github-actions/expressions#about-expressions" rel="noopener nofollow ugc">expression syntax</a>:</p>
<pre><code class="lang-auto">if: ${{ !contains(github.event.pull_request.labels.*.name, '0 diff trivial') }}
</code></pre>

**Reply by rffontenelle** [reply]
Awesome. When running a multi-line !contains statement, surrounding with double quotes raised syntax error in the editor. But your suggestion worked just fine.

**Comment by skoonin** [comment]
for future people, i struggled getting this to work with multiple labels.. the format is: 

```
if: "!(contains(github.event.pull_request.labels.*.name, 'label1') || contains(github.event.pull_request.labels.*.name, 'label2'))"
```

**Reply by DanyC97** [reply]
@skoonin  thank you so much for sharing the snippet, i spent huge amount of time trying to find out why the heck was not working. Am extremely angry, how much effort will be for a GH eng to update and add a minor example on this one?  

thx again 🙌
